### PR TITLE
Added support for using ControlTipText and Caption fields.

### PR DESCRIPTION
**Pull request by kirk-sayre-work**
Added support for reading ControlTipText field from OleSiteConcreteControl.

Made a small change to save the Caption text returned from consume_LabelControl in the dict returned from extract_OleFormVariables. A field for ControlTipText was also added to this dict.

This change was motivated by maldocs like 8882fb51cdbbd860ddaaabec98603a5f2dd700fdebabab12000202fc79cb609c .

### CODE REVIEW

**Comment by kirk-sayre-work** [comment]
I just pushed a fixed version of the oleform unit tests to take into account the new fields.

**Comment by kirk-sayre-work** [comment]
It looks like there might be an off byte 2 byte error when reading in the form tag following the form name. As an example check out maldoc 74ab169048e1ab22caeb167ff4cbce34ea845d9e159a029c881ec48936f7770a .

**Comment by decalage2** [comment]
The tests only fail with Python 2.7 and pypy 2, it's OK for Python3.x:
```
======================================================================
FAIL: test_samples (tests.oleform.test_basic.TestOleForm)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/decalage2/oletools/tests/oleform/test_basic.py", line 27, in test_samples
    self.assertEqual(variables, expected_result)
AssertionError: Lists differ: [('word/vbaProject.bin', u'Use... != [('word/vbaProject.bin', u'Use...
First differing element 0:
('word/vbaProject.bin', u'UserFormTEST1', {'name': 'Label1', 'tag': 'sdfff', 'id': 1, 'tabindex': 0, 'ClsidCacheIndex': 21, 'value': None, 'caption': 'Label1-test', 'control_tip_text': None})
('word/vbaProject.bin', u'UserFormTEST1', {'caption': 'Label1-test', 'control_tip_text': None, 'name': 'Label1', 'value': None, 'tag': 'l\x18sdf', 'ClsidCacheIndex': 21, 'id': 1, 'tabindex': 0})
Diff is 5136 characters long. Set self.maxDiff to None to see it.
```

**Comment by decalage2** [comment]
Hi @kirk-sayre-work, I'd like to process this PR, but it seems to remove features that I added recently (VBA stomping detection). Could you please rebase, so that it does not affect existing code?

**Comment by kirk-sayre-work** [comment]
I just merged from your master branch. Did that fix the problem?

**Comment by decalage2** [comment]
Hi @kirk-sayre-work, unfortunately this didn't fix the issue. I checked all the last commits, and this one seems to cause the issue, removing the code that I added to detect VBA stomping: 4188b3d137c358ecb4e80e4f78424b609d7a23ac

File: oletools/olevba.py [hunk 1]
```diff
@@ -2627,8 +2627,11 @@ def __init__(self, filename, data=None, container=None, relaxed=False, encoding=
             # and even whitespaces in between "MIME", "-", "Version" and ":". The version number is ignored.
             # And the line is case insensitive.
             # so we'll just check the presence of mime, version and multipart anywhere:
-            if self.type is None and b'mime' in data_lowercase and b'version' in data_lowercase \
-                and b'multipart' in data_lowercase:
+            if (self.type is None and
+                b'mime' in data_lowercase and
+                b'version' in data_lowercase and
+                b'multipart' in data_lowercase and
+                abs(data_lowercase.index(b'version') - data_lowercase.index(b'mime')) < 20):
```
Comment (decalage2) [inline oletools/olevba.py]: Hi Kirk, do you have a sample where this was an issue?

**Comment by kirk-sayre-work** [comment]
There is a Trickbot VBS dropper that generates giant comment lines from a random word list that results in the substrings ‘mime’, ‘version’, and ‘multipart’ being scattered throughout the VBS file, which results in olevba classifying the file as MHT, which then causes olevba to say there are no macros in the file. Here is a list of example hashes (all on VT):

03d6c7c5ba7cfbc8b5f1bea2d2b1ff03e7f95bf4d7a2d525977c6ded02dc5e78
047932c8847bda0a6397ab90ec1729e18d6a1b1bc5fb83a783048a9c68efb91c
049a76ee69fb87a12a11a89471931f28df123154e8ba6d20c3fe844cb625fe6e
082fc4bcf863b27da41e45074908f8875be647f672bb8f619f16e7a4e7db3ec7
0eefeb632f04453518919c37baa49c398315eaf330c1b24dbace1a5e6e77dadc
10c9831dd60186d8fd80bef2ebc30cd71451eca89772ef3ec392fb2e180109dc
1198a829c071e8238c222d8ceafb283a57e0eaf6b8a2f9e6a09c35c88478ddb9
1428c22539fc370318408f3bf23ac97b1f60d8e3fd5f7cf19b66ec382d569050
144b915fd006f4f20b34ca068f2fce18277b7e914470caffd65a9e6929d37167
166fd8be987a2faf7e22ce8a0606e6a59bc7230951114173e1d98fbb066370d8
1e0b966d66dc7333ac674ac3e6c7c3c5e6f3518e5822cdbefd945370d9511f53
1fa92bb324929661dc6b9de7c773d534249e0cc8c6cd6dbbdbde899f1a7f92d9
21630accc67dacbfef3f49a720ed9cac3919b68738b300dc6a13975da8895cb0
272dc1556122ee1a160f4688a376ec0ccb33376f0801826acb5d5cd7c6208dbc
28372d0619a6d5299d98b6aca5bc743e4ed0a4529e17835fd2883769e2918527
292fd3b78e2b8f995f8513decb3d71861a46b2cf266fa07f47f577c76260d249
391f540e2a4cd2d4f6bcbfb0da895f6b37c5b89e591167181189fcbf3a4080a1
3a29cd44d9ba06ac29fe8ad1f0ac00705549c64db282c30751aa21260ecb0119
3d3741d18788a93cbeaef1c0cf0fda23519eee083e9d5b832013040476729c9c
3f2a1b23c944a9f1d8ca8339408ea90597b0849338a5bd4dfe5100a28bda8ffa
3fb9dc9ed3f8200146fd584b593bdd02775eb9c11f58111d434ca9182a988315
411d441cb0c52d39a4edb6ed186acadb99ae04909e6da25f86c48d2adf0550f6
41e1e4059f70d771cae26309b20674cf64b69a09f80ae92120ccf54966b299f5
442b9439ee6a7e732b6668d4b878ce4d56dced249970a3f9a579a8f030d9d495
44b9150ca68cdd74ee41d46e503540498751bc973da21da4a22468ac2b865fe6
45fe73b42714f91649e5bebe0ef45849c2cfa70329f6b28dd76df68f1e5fdfb4
4e2d9a168f671b9fe1b7e0ce605dca24b6d4419890e3f20bb322f494b91e5e48
53c0034c0b84b5788dd78d7965d3b91b7a86e801fa296b9e01aa45adaf7ce153
55f4fb862882e7d2654c08eeaf555ef578506b92d656de074952e84c3199062d
58ff9732ba021a66a3cae7db6f4bd334fb3b32bbd9d26a67450a7e9d8330410a
5a66e75f5f1516d756f91db7e67b1ab8c9e7acfea51c1578e76450e3b23563b4
5c586456dfbaea32b63c0f2cceaef150301a68cbb6f05141c5839e4c83118e91
5d72379e0a75e8c66efd5a9b1634e5cab1636a40609f614edfb7dca2969fd46e
651927967bd05bf3a8b44fb58d38089c5a3cf758c3a0affa7a8342d9934cc017
68e3eaddfbf26eb6285f314fcfd843d4bea9d54a18d056b814f5b94c963d6b4b
6f167727722933e3698e660e5db9ffe448c30863a77ef5977bb11229f82ecc30
6f4a36d71527e08a17e0eba6c87284f41cff34b6c428aceea36cb05b75c3ae78
755722e4ea5e89152e210f1f19cbbccbbf42648b251b4e236a78136881877323
78b6cb00b13ee08585e7b8b1717bbd0c78dfbb973542ba9212ae592bb92ac8f0
7cb08cabae40f72fd52683e9e9adf4328e65dac420a6b075f57bd221ee7f8c06
7fabf89314049f5bebe360929d01c6db6b21d9ec092e696c6c8cfc89bd69cc9f
839a230aa4d4ce1566ae0aef48f281509ff8ffabe7baf3edd6b2c1c15e5fc3ec
85fd673c3e4c0f8753bef9f249ff31f80b60bb62a42db7364023a9077aa81fc0
889be8cc360a9b867311996f0e805cc6caa96a7e9de0d94b2bab829611e21703
8a19019a754b36487e445ed1f9ea653adb16df210bca00fd8458c29a5c7f7c7c
8cecf224bedc4bdf36e4d823d431755b51fc02340ccb3b2a7745cd16664ebd26
93e7da9af41cc7ae1353e83526ec51ec5b42a70d7a70f5f4d239334d3da80ebf
96af5a843e57220476c9b7f5f39a75f3e74c10711c71349e285295f84436be3d
9c56ac8743a1a1bf0d5c6125e874c0a380b2cab0b9d979cfe2c77a4b1a7b2510
a420f4891b512ea09f5dcb2d976f2be1b4aaf098ee5d068659107e5cf9d3818e
a9381cd1040126835e4a1e28bf367a7af92a9ab92fbe566e80e67d177dd0b1b5
ab27d50c4b50a392c10f05751cf5e0f4aae29a4d0000eeaf9ce4184f5339a591
b43d4ca21f2a4e8a8b11cbf9af82a7b5de5e9cd341670f5973a05c59ed2eff29
b4aaaf2b2da0d354c34f30da4428ac2275749439bec0a4bb13141fd81f09cf1e
b60b23ce32d022a315c59708bb8d006779700950099154c64efb622c862e6453
b9f692a226ab493ed89e208a5da6e30dbba47b53df657f2c732082deeca99663
bf603414d36ffd4fc5eea4997dac8e32e2ddac6753a00de74c43ba659b53dc2c
bfded4e782803fe1827ffa4dd6c3b70856680b1838ee79811ec73fbd220f6478
c2136b867c530fac781836e71987f56c1fcc701ac4361189e1b1a50ed1153c8f
c31e65a3e7a0a952d248f4b2a2882990f1734e123879122c5f485f81f6fe4bb5
c3d15910e275f1e8e8b0c9bf82aec3ec39e5b3ec59d6edfe8f4fa5731f08784d
c906a5d2971ab875ef44487646be513b8d179fe747b442841efe508d04b6f0cb
cb65366989a240b39a3eedbcb1be78ef19a6136341725b5c58d0b3d0a76b1254
cb834d6e43bb1a301ca06815f83e26ae7c0e51af44a849574aed75efab6f9437
cc13b87dfb342755d7871c5c4eeb93bac5e9f4ba1f807c569330504be0d0280c
d106c6afba88309f3dac8976e04274898b899d494262f3a182a502b5625860a2
d1bf6fef7e5080453b1b9a17fdc5ba61e089404b945860093439429ed7a639d0
d66331dee45b7b38e42976a8e6195ede51e92ff629db119e08cddedc75c6c091
ddae11922748b03e197284ed75a34abbfa5d67a4f8a97b118b4057d9fbca2359
dfc311266a1d577f5ff64eee8030099dec1676270bc105d5e7de3218f3364660
e4b57c885637e443d1cdcf1cc077f2fa9ef7fb87a7f4c8eb9d47889973aaed5f
ee4957be85d774e367d76c4d901f8717c4ed509eed3f77afb97edf3d81a66429
f18a1a581ebe8b8fa01f8a43bf41986f24625125b9dd157b02edc54569938cd5
f43a641d4801a196510a956924420bba114bd5a81a72f095f19e0541d1dc18ab
f878fcdfa6d56310dc1782fc164e066d62e6bd5d7e72d723d33758f6e83a6035
f95b8f74a087432b63602214aa484b0d74b30993fc38f72d0c27045d7f69284c
f9cee2b598f8f6710930c49877792d92daf9c96a452fd274e565ca0be2c4b13c
fb0be8cff85fe1774f4b2b5dd47f842dff604c7eaba6254c60b0802cace59440
fb5c237e79b6da8ab9b878a2176fcde8912454d145798a645d544e101cbbb594
fe9f44e79c2c01de7d4bb6c7adf0efa75f9e8fa2951e8ae3e561e112c526f5e8
ff1a4a6780c862f6b56be12a5f2787b5f1c7478a6701b7d3dbb77ee17bc98a44
ff5ef11b9f912dcf8f714a6363f1f243e10236ff0d7bb3984d294477c7322974

Kirk Sayre
Dynamic Defense Engineering Team

**Pull request closed without merging**